### PR TITLE
Update `gravitational/protobuf` fork tag

### PR DESCRIPTION
**Pull request by jentfoo**
Previously we were not referencing a tag, instead we were referenced to a commit on this branch: https://github.com/gravitational/protobuf/tree/dmitri/merge-values

With our fork referencing version `1.3.2` it was thought that this branch contained the fix for the `Skippy Peanut Butter` CVE: https://nvd.nist.gov/vuln/detail/CVE-2021-3121

However that was not the case, so the following was performed:
* A `teleport` branch was created on `protobuf` from the patched upstream v1.3.2 tag
* Our custom commits were cherry-picked in (the Clone Merger functionality added, codeql config, and dependency-review config)
* It was pushed as tag `v1.3.2-1` to our fork (as seen in this change)

Thank you @espadolini for helping highlight this issue

### CODE REVIEW

**Review by codingllama — APPROVED** [review]
LGTM for local changes (haven't reviewed the fork itself).

**Comment by jentfoo** [comment]
Full fork change set can be seen here: https://github.com/gravitational/protobuf/compare/dmitri/merge-values...gravitational:protobuf:teleport

File: go.mod [hunk 1]
```diff
@@ -389,7 +389,7 @@ replace (
 	github.com/datastax/go-cassandra-native-protocol => github.com/gravitational/go-cassandra-native-protocol v0.0.0-20221005103706-b9e66c056e90
 	github.com/go-mysql-org/go-mysql => github.com/gravitational/go-mysql v1.5.0-teleport.1
 	github.com/go-redis/redis/v9 => github.com/gravitational/redis/v9 v9.0.0-teleport.3
-	github.com/gogo/protobuf => github.com/gravitational/protobuf v1.3.2-0.20201123192827-2b9fcfaffcbf
+	github.com/gogo/protobuf => github.com/gravitational/protobuf v1.3.2-1
```
Comment (wadells) [inline go.mod:392]: I might even go with a more explicit prerelease tag like `v1.3.2-teleport.1` to make it clear this is our work.

**Review by wadells — APPROVED** [review]
LGTM.

Similar to @codingllama's caveat, I can't say I've looked at the knock on effects, but I approve of the spirit of this change.

File: go.mod (hunk 1 above)
Comment (jentfoo) [reply go.mod]: Good recommendation, I see this pattern being used in the replacements above.  I have updated the tag to `v1.3.2-teleport.1`

**Review by codingllama — APPROVED** [review]

**Review by espadolini — APPROVED** [review]
From what I can tell, the only change is actually in the codegen, and we were already using the latest version of the codegen, so this should not change functionality.

**Pull request merged**